code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -199,18 +199,19 @@
 		ui.open()
 
 /mob/living/simple_animal/bot/cleanbot/ui_data(mob/user)
-	var/list/data = list(
-		"locked" = locked, // controls, locked or not
-		"noaccess" = topic_denied(user), // does the current user have access? admins, silicons etc can still access bots with locked controls
-		"maintpanel" = open,
-		"on" = on,
-		"autopatrol" = auto_patrol,
-		"painame" = paicard ? paicard.pai.name : null,
-		"canhack" = canhack(user),
-		"emagged" = emagged, // this is an int, NOT a boolean
-		"remote_disabled" = remote_disabled, // -- STUFF BELOW HERE IS SPECIFIC TO THIS BOT
-		"cleanblood" = blood
-	)
+	var/list/data = list()
+	data["locked"] = locked // controls, locked or not
+	data["noaccess"] = topic_denied(user) // does the current user have access? admins, silicons etc can still access bots with locked controls
+	data["maintpanel"] = open
+	data["on"] = on
+	data["autopatrol"] = auto_patrol
+	data["painame"] = paicard ? paicard.pai.name : null
+	data["canhack"] = canhack(user)
+	data["emagged"] = emagged // this is an int, NOT a boolean
+	data["remote_disabled"] = remote_disabled
```

Comment:
Actually, if you're refactoring this, can you put all of this on /bot's ui_data, and then just call parent with `var/list/data = ..()`? It would cut down on a lot of this for all of these procs